### PR TITLE
Rename Auditbeat modules

### LINKED ISSUE #5422: Refactoring Auditbeat Modules
We want to do some refactoring to change the config and event schema before the Auditbeat GA release.  Auditbeat reuses Metricbeat's framework, and the framework's orientation around metrics is visible to the Auditbeat user, but it shouldn't be.

### Goals

- Simplify configuration for the end-user
  - Modules should map clearly to events the users want to monitor.
- Divorce from the Metricbeat event data model
  - The `metricset` namespace isn't necessary here.
- Abstract the data source from module
  - Users shouldn't need to know how the data is collected
  - If needed an advanced user should be able to control what data sources are used.
  - For example have a module for monitoring what processes are running or what sockets are active rather than a module oriented around a data source like the kernel's audit framework or eBPF.
- Provide specialized dashboards for each module.

### Steps

1. Modify the Metricbeat framework to allow for customization of the event builder (the builder is responsible for adding the `metricset` field to all events).
1. Reorganize the config to allow for modules without "metricsets".
1. Rename the existing audit/kernel metricset to the `auditd` module. (As we add modules I expect this to be used less.)
1. Rename the existing audit/file metricset to `file_integrity`.

### Sample Configs

Note the removal of `metricsets: []` and the removal of the module name prefixes from the config options.

```
auditbeat.modules:
- module: file_integrity
  paths:
  - /bin
  - /usr/bin
  - /sbin
  - /usr/sbin
  - /etc
  scan_at_start: true
  scan_rate_per_sec: 50 MiB
  max_file_size: 100 MiB
  hash_types: [sha1]
```

```
auditbeat.modules:
- module: auditd
  resolve_ids: true
  failure_mode: silent
  backlog_limit: 8196
  rate_limit: 0
  include_raw_message: false
  include_warnings: false
```

### PR DESCRIPTION
**Pull request by andrewkroh**
This changes the audit.file and audit.kernel metricsets into modules
named file_integrity and auditd, respectively. This requires existing
users to update their configuration.

The dashboards need to be updated to account for these changes.

Closes #5422 (see the issue for more details)

### CODE REVIEW

**Comment by exekias** [comment]
jenkins retest it, please

File: Vagrantfile [hunk 1]
```diff
@@ -119,16 +122,25 @@ Vagrant.configure(2) do |config|
...
-    centos7.vm.provision "shell", inline: $unixProvision, privileged: false
-    centos7.vm.provision "shell", inline: $linuxGvmProvision, privileged: false
+    c.vm.provision "shell", inline: $unixProvision, privileged: false
+    c.vm.provision "shell", inline: $linuxGvmProvision, privileged: false
 
-    centos7.vm.synced_folder ".", "/vagrant", type: "virtualbox"
+    c.vm.synced_folder ".", "/vagrant", type: "virtualbox"
+  end
+
+  config.vm.define "fedora26", primary: true do |c|
```
Comment (exekias) [inline Vagrantfile]: is this being used by beats-tester or just for development?

**Comment by exekias** [comment]
Have a look to https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-docs/2505/console, it seems docs tests are failing

File: Vagrantfile (hunk 1 above)
Comment (andrewkroh) [reply Vagrantfile]: It's not used by beats-tester. I had been using it for testing so I figured might as well add it.

File: auditbeat/_meta/fields.common.yml [hunk 2]
```diff
@@ -3,34 +3,7 @@
   description: >
     Contains common fields available in all event types.
   fields:
-
-    - name: metricset.module
+    - name: dataset.module
```
Comment (ruflin) [inline auditbeat/_meta/fields.common.yml]: Interesting 👍

File: auditbeat/docs/modules/auditd.asciidoc [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+- module: auditd
+  resolve_ids: true
+  failure_mode: silent
+  backlog_limit: 8196
+  rate_limit: 0
+  include_raw_message: false
+  include_warnings: false
+----
+
+*`socket_type`*:: This optional setting controls the type of
```
Comment (jarpy) [inline auditbeat/docs/modules/auditd.asciidoc]: `socket_type` option is missing from the list, above.

File: auditbeat/docs/modules/auditd.asciidoc [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+primarily used for development and debugging purposes.
+
+*`include_warnings`*:: This boolean setting causes {beatname_uc} to
+include as warnings any issues that were encountered while parsing the raw
+messages. The default value is false. When this setting is enabled the raw
+messages will be included in the event regardless of the
+`include_raw_message` config setting. This setting is primarily used for
+development and debugging purposes.
+
+*`audit_rules`*:: A string containing the audit rules that should be
```
Comment (jarpy) [inline auditbeat/docs/modules/auditd.asciidoc]: `audit_rules` is not in the list of all options, above.

File: auditbeat/docs/modules/file_integrity.asciidoc [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+[source,yaml]
+----
+- module: file_integrity
+  paths:
+  - /bin
+  - /usr/bin
+  - /sbin
+  - /usr/sbin
+  - /etc
+  scan_at_start: true
```
Comment (jarpy) [inline auditbeat/docs/modules/file_integrity.asciidoc]: This section implies that `paths` has a default value, but if you don't declare it yourself, you get:

```
2017/12/22 07:09:57.004627 beat.go:644: CRIT Exiting: 1 error: 1 error: missing required field accessing 'auditbeat.modules.1.paths' (source:'auditbeat.yml')
Exiting: 1 error: 1 error: missing required field accessing 'auditbeat.modules.1.paths' (source:'auditbeat.yml')
```

**Pull request merged**